File: tasks/champernowneDigit.js
function champernowneDigit(n) {

  for (var i = 1;; i++) {
    var number = i,
      digits = [];
    while (number) {
      digits.push(number % 10);
      number = Math.floor(number / 10);
    }
    digits.reverse();
    if (n <= digits.length) { //bugfix
      return digits[n - 1];
    }
    n -= digits.length;
  }
}

//CodeWriting
function champernowneDigit(n) {

  var s = "";
  for (var i = 0; i < 1000; i++) {
    s += i + 1;
  }
  console.log(s);
  return parseInt(s[n - 1])
}

File: tasks/champernowneDigit.py
def champernowneDigit(n):

    i = 1
    while True:
        number = i
        digits = []
        while (number > 0):
            digits.append(number % 10)
            number /= 10
        digits.reverse()
        if n <= len(digits):
            return digits[n - 1]
        n -= len(digits)
        i += 1
